### PR TITLE
[fix/docs]: fit `check_amicable_pair.cpp` to guidelines

### PR DESCRIPTION
**Pull request by ewd00010**
#### Description of Change
FIX:  initlized sum to 1 instead of adding it before return
CHORE: cleaned documentation aswell as adding new documentation, namespace math added

#### Checklist
- [x] Added description of change
- [x] Added tests and example, test must pass
- [x] Added documentation so that the program is self-explanatory and educational - [Doxygen guidelines](https://www.doxygen.nl/manual/docblocks.html)
- [x] Relevant documentation/comments is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#Commit-Guidelines)
- [x] I acknowledge that all my contributions will be made under the project's license.

Notes: could use clang-tidy check like my previous pull requests

### CODE REVIEW

File: math/check_amicable_pair.cpp [hunk 1]
```diff
@@ -1,26 +1,36 @@
...
+ * divides into a selected number, apart from the selected number itself, and
+ * returns a positive integer. for example 1, 2 and  5 are all proper divisors
+ * of 10.
+ *
+ * @author [iamnambiar](https://github.com/iamnambiar)
  */
-#include <cassert>
-#include <iostream>
+#include <cassert>   // for assert
+#include <iostream>  // for cout
```
Comment (realstealthninja) [inline math/check_amicable_pair.cpp:18]:
```suggestion
#include <cassert>   /// for assert
#include <iostream>  /// for cout
```

**Review by realstealthninja — CHANGES_REQUESTED** [review]

**Review by realstealthninja — APPROVED** [review]
Thank you for your pr. We greatly appreciate your help here. Now the maintainers will review your pr! ❤️

File: math/check_amicable_pair.cpp [hunk 2]
```diff
@@ -48,25 +58,27 @@ int sum_of_divisor(int num) {
...
+ * @returns void
  */
-void test() {
-    // are_amicable(220, 284) returns true.
-    assert(are_amicable(220, 284) == true);
-    // are_amicable(6232, 6368) returns true.
-    assert(are_amicable(6368, 6232) == true);
-    // are_amicable(458, 232) returns false.
-    assert(are_amicable(458, 232) == false);
+void tests() {
```
Comment (Panquesito7) [inline math/check_amicable_pair.cpp:67]:
```suggestion
static void tests() {
```

File: math/check_amicable_pair.cpp [hunk 3]
```diff
@@ -48,25 +58,27 @@ int sum_of_divisor(int num) {
...
+    assert(math::are_amicable(220, 284) == true);
+    assert(math::are_amicable(6368, 6232) == true);
+    assert(math::are_amicable(458, 232) == false);
+    assert(math::are_amicable(17296, 18416) == true);
+    assert(math::are_amicable(18416, 17296) == true);
+
+    std::cout << "all tests have passed" << std::endl;
```
Comment (Panquesito7) [inline math/check_amicable_pair.cpp:74]:
```suggestion
    std::cout << "All tests have successfully passed!" << std::endl;
```

**Review by Panquesito7 — CHANGES_REQUESTED** [review]

File: math/check_amicable_pair.cpp [hunk 4]
```diff
@@ -1,26 +1,36 @@
...
+ * divides into a selected number, apart from the selected number itself, and
+ * returns a positive integer. for example 1, 2 and  5 are all proper divisors
+ * of 10.
+ *
+ * @author [iamnambiar](https://github.com/iamnambiar)
  */
-#include <cassert>
-#include <iostream>
+#include <cassert>   /// for assert
+#include <iostream>  /// for cout
```
Comment (Panquesito7) [inline math/check_amicable_pair.cpp:18]:
```suggestion
#include <iostream>  /// for IO operations
```

**Review by realstealthninja — DISMISSED** [review]
LGTM! great work

**Review by Panquesito7 — DISMISSED** [review]
Awesome. Thanks! 🚀

**Review by realstealthninja — DISMISSED** [review]
Great work

**Review by Panquesito7 — APPROVED** [review]

**Pull request merged**